pkg/handler/eval_cache.go

```diff
@@ -59,6 +63,39 @@ func (ec *EvalCache) Start() {
 	}()
 }
 
+func (ec *EvalCache) GetByTags(tags []string) []*entity.Flag {
+	ec.mapCacheLock.RLock()
+	defer ec.mapCacheLock.RUnlock()
+
+	results := []*entity.Flag{}
+	for _, t := range tags {
+		s := util.SafeString(t)
+		f, ok := ec.tagCache[s]
+		if ok {
+			results = append(results, f...)
+		}
+	}
+
+	return FlattenFlags(results)
```

Comment:
I can see that you are doing dedup here. Since flag id order doesn't matter, how about we define 

```
tagCache:        map[string]map[uint]*entity.Flag      // i.e. tag => map[flagID]*entity.Flag
```

And when loop through the tags, we just need to get a union of the maps, and then return the values of that union map as a slice